website/frontend/src/components/pages/Programs/children/Upload/children/CodeModal/CodeModal.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useAlert } from 'react-alert';
+import { Hex } from '@gear-js/api';
+import { ISubmittableResult } from '@polkadot/types/types';
+import { web3FromSource } from '@polkadot/extension-dapp';
+import copy from 'assets/images/copy.svg';
+import { copyToClipboard, readFileAsync } from 'helpers';
+import { useApi } from 'hooks/useApi';
+import { EventTypes } from 'types/alerts';
+import { RootState } from 'store/reducers';
+import { AddAlert } from 'store/actions/actions';
+import { Button } from 'common/components/Button/Button';
+import { Modal } from 'components/blocks/Modal';
+import { Spinner } from 'components/blocks/Spinner/Spinner';
+import { DroppedFile } from '../../types';
+import styles from './CodeModal.module.scss';
+
+type Props = {
+  file: File;
+  setDroppedFile: Dispatch<SetStateAction<DroppedFile | null>>;
+};
+
+const CodeModal = ({ file, setDroppedFile }: Props) => {
+  const [api] = useApi();
+  const alert = useAlert();
+  const dispatch = useDispatch();
+  const { account } = useSelector((state: RootState) => state.account);
+  const [codeHash, setCodeHash] = useState('' as Hex);
+
+  const close = () => {
+    setDroppedFile(null);
+  };
+
+  const handleClick = () => {
+    copyToClipboard(codeHash, alert);
+    close();
+  };
+
+  const handleSuccess = (hash: Hex) => {
+    setCodeHash(hash);
+    dispatch(AddAlert({ type: EventTypes.SUCCESS, message: 'Code uploaded' }));
+  };
+
+  const handleFail = (message: string) => {
+    close();
+    dispatch(AddAlert({ type: EventTypes.ERROR, message }));
+  };
+
+  const submit = async () => {
+    const arrayBuffer = (await readFileAsync(file)) as ArrayBuffer;
+    const buffer = Buffer.from(arrayBuffer);
+    return api.code.submit(buffer);
+  };
+
+  const handleUpload = ({ events }: ISubmittableResult, hash: Hex) =>
+    events.forEach(({ event: { method } }) => {
+      if (method === 'CodeSaved') {
+        handleSuccess(hash);
+      } else if (method === 'ExtrinsicFailed') {
+        handleFail('ExtrinsicFailed');
```

Comment:
Display text of errors here. After this https://github.com/gear-tech/gear-js/pull/390 will be merged